lib/internal/errors.js

```diff
@@ -1122,7 +1122,26 @@ E('ERR_OUT_OF_RANGE',
     msg += ` It must be ${range}. Received ${received}`;
     return msg;
   }, RangeError);
-E('ERR_REQUIRE_ESM', 'Must use import to load ES Module: %s', Error);
+E('ERR_REQUIRE_ESM',
+  (filename, parentPath = null, packageJsonPath = null) => {
+    const path = require('path');
+    const basename = path.basename(filename) === path.basename(parentPath) ?
+      filename : path.basename(filename);
+    let msg = `Must use import to load ES Module: ${filename}\n`;
```

Comment:
The newline at the end here should be moved to the beginning of the branch rather - `msg += '\nrequire() ...`